### PR TITLE
Document required IAM permissions for S3 storage driver.

### LINKED ISSUE #1391: Better documentation for S3 driver IAM permissions
As mentioned in #492, the s3 driver requires the following minimal permission scope to operate correctly when configured to use the bucket `mybucket` at the root:

```json
    {
        "Effect": "Allow",
        "Action": "s3:ListAllMyBuckets",
        "Resource": "arn:aws:s3:::*"
    },
    {
        "Effect": "Allow",
        "Action": [
            "s3:ListBucket",
            "s3:GetBucketLocation"
        ],
        "Resource": "arn:aws:s3:::mybucket"
    },
    {
        "Effect": "Allow",
        "Action": [
            "s3:PutObject",
            "s3:GetObject",
            "s3:DeleteObject"
        ],
        "Resource": "arn:aws:s3:::mybucket/*"
    }
```

This should be added to the s3 driver documentation, as well as further information regarding a more limited scope when using a bucket "root directory" prefix.

### PR DESCRIPTION
**Pull request by RichardScothern**
Document required IAM permissions for S3 storage driver.

Closes #1391 
Signed-off-by: Richard Scothern <richard.scothern@docker.com>

### CODE REVIEW

**Comment by RichardScothern** [comment]
cc @drewrobb
cc @nwt

**Comment by drewrobb** [comment]
Thank you, this looks good to me.

File: docs/storage-drivers/s3.md [hunk 1]
```diff
@@ -187,6 +187,42 @@ An implementation of the `storagedriver.StorageDriver` interface which uses Amaz
...
+## S3 permission scopes
+
+The following IAM permissions are required by the registry for push and pull.  See [the S3 policy documentation](http://docs.aws.amazon.com/AmazonS3/latest/dev/mpuAndPermissions.html) for more details.
+
+```
+ "Statement": [
+      {
+        "Effect": "Allow",
+        "Action": [
+          "s3:ListAllMyBuckets"
```
Comment (aaronlehmann) [inline docs/storage-drivers/s3.md]: Why does the registry need this?

**Comment by aaronlehmann** [comment]
Thanks for this. This will be a huge help for users.

File: docs/storage-drivers/s3.md (hunk 1 above)
Comment (dmp42) [inline docs/storage-drivers/s3.md]: Agreed we should investigate.

Comment (RichardScothern) [inline docs/storage-drivers/s3.md]: `s3:ListAllMyBuckets` is needed to view the objects in a console.  I wonder if, in the original, contribution (#492) that the IAM was used by the registry and the user.

Comment (dmp42) [inline docs/storage-drivers/s3.md]: Ok. can any of you gents (@drewrobb @nwt) check if this is really required to run?

Comment (RichardScothern) [inline docs/storage-drivers/s3.md]: or maybe @BrianBland (if he's out there?) :grin:

Comment (drewrobb) [inline docs/storage-drivers/s3.md]: I thought I had commented here earlier. I tested and it and I didn't get errors without it. However I do remember getting errors without it on prior versions of registry, so I would wait for a second opinion here.

Comment (RichardScothern) [inline docs/storage-drivers/s3.md]: I will remove this - it does seem overly permissive and a couple of people have reported it's not required for aws-sdk

**Comment by aaronlehmann** [comment]
LGTM

File: docs/storage-drivers/s3.md (hunk 1 above)
Comment (BrianBland) [inline docs/storage-drivers/s3.md]: I don't see any reason why `ListAllMyBuckets` would be needed with the current implementation. The same goes for `GetBucketLocation`, but everything else looks correct.

**Pull request merged**